exist-docker/src/main/resources-filtered/Dockerfile

```diff
@@ -20,25 +20,7 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
 #
 
-# Use Debian Bullseye (which is the base of gcr.io/distroless/java:17) for additional library dependencies that we need
-# FROM debian:bullseye-slim as debian-slim
-# RUN apt-get update && apt-get -y dist-upgrade
-# RUN apt-get install -y openjdk-17-jre-headless
-# RUN apt-get install -y expat fontconfig     # Install tools required by FOP
-
-FROM gcr.io/distroless/java17:latest
-
-# Copy over dependencies for Apache FOP, missing from GCR's JRE
```

Comment:
This change breaks XSL:FO in eXist-db when using Apache FOP. Unfortunately, the `gcr.io/distroless/java17-debian12:latest` image you have chosen to update to does not include these libraries. Please test a proper FOP transformation with these changes, and you will see the error.

Reply:
This PR switches from  `gcr.io/distroless/java17:latest` to `gcr.io/distroless/java17-debian12:latest`
The lines stating they would be needed for FOP are commented out (lines 24-27 and 32-41) and select a different base image `debian:bullseye-slim`.
So, even if those additional libraries were needed for PDF-generation using XSL:FO they are already missing and nothing is changed from the status quo.  
It, therefore, would be beneficial to create an additional Dockerfile that has all that is needed for this specific use-case. This could also be provided by a third party.
This is all just under the pretense that this is really needed for PDF generation and the adoption rate would also show us how common it is.

Reply:
If they are no longer included that's a bug likely introduced when moving to debian 12. see https://github.com/GoogleContainerTools/distroless/issues/321 or https://github.com/GoogleContainerTools/distroless/issues/789 from 2019 and 2021 for relevant discussions. I will check the java-base image and move the discussion to the distroless repo

Reply:
I don't see it as a bug with Debian 12 or GCR. Simply, GCR has never included these libraries as many Java Applications don't need them. However, we do need them for eXist-db, which is why previously explicitly installed them into a base Debian image and then copied them into our derivative of GCR.

Reply:
@adamretter they were included in the gcr java-base image.

Reply:
@duncdrum I can't see it in the GCR java-base images (see below)... Where are you seeing it?

<img width="841" alt="Screenshot 2024-11-21 at 15 43 57" src="https://github.com/user-attachments/assets/7de40252-ee67-45e8-aa80-f059a95a3e84">

<img width="842" alt="Screenshot 2024-11-21 at 15 41 33" src="https://github.com/user-attachments/assets/36960067-30c1-4e58-b7f8-f861be66fae6">

Reply:
@duncdrum Interesting that the files even though listed in that BUILD file don't seem to have ever been in the GCR images as far as I can see. Also, I think even if they were there, from what I understand of that BUILD file ,they would still be missing libexpat which we need for Apache FOP (via Apache Batik)

Reply:
Just to be sure @adamretter    You are checking in subsequent layers as well, and not seeing the dependencies?

Reply:
> You are checking in subsequent layers as well, and not seeing the dependencies?

@duncdrum Yes, I think so. What do you see when you look in the image?

Reply:
@adamretter now that I'm back at my laptop. On the image published as part of this PR I see all the dependencies present in the base image, as would be expected based on the build file.
<img width="1006" alt="Screenshot 2024-11-25 at 19 35 36" src="https://github.com/user-attachments/assets/aca6dbbc-c8a4-47ed-a85b-25e7e8ead83c">

Reply:
@adamretter nope WAI. There was some discussion about the reasoning behind separate vs single layers in distroless slack. The overall size hasn't changed much, so any differences should only be noticeable on the first download.

Reply:
> WAI

What does "WAI" mean?

Reply:
Having more layers means they might be better cached, I suppose.

Reply:
I think that would only work if your lower levels are stable, i.e. don't change. As this now proposes using a `latest` as the base image, that means any change from that explodes upwards and changes all of our upper layers, so therefore they are cache invalidated.

Reply:
@adamretter WAI mean working as intended. There are pro and cons to layer squashing. The distroless images now come with more layers. `latest` was used before and after this PR. There is no change to cache busting or change inheritance mechanism.